### PR TITLE
method signature lookups, parameter decoding & management

### PR DESCRIPTION
**Pull request by jacogr**
There are 2 parts to this -

- dapp for on-chain signature management. (PR open for that under contracts as well)
- initial lookup, decoding and display of transaction methods against the signature registry (only shows up on transactions view currently)

Future work -

- make methods display & parameters look nice, not “techy”
- shared component for this to be added to Signer (consistent view, all transactions in the app show the decoded details)

Demos -

- dapp for registration https://www.youtube.com/watch?v=Z8z4qdxnSn0
- signature lookups & decoding in UI (EDIT: updated l&f) https://youtu.be/2R45Ho_Y9Pk (original: https://www.youtube.com/watch?v=BM3lL6hQhdo)
- semantic analysis first-stab - https://www.youtube.com/watch?v=z_02UwdjzHY&feature=youtu.be

Linked issues -

- Signature lookup https://github.com/ethcore/parity/issues/2208
- Semantic analysis - https://github.com/ethcore/parity/issues/2280

Dependencies -

- Will stay WIP until https://github.com/ethcore/parity/pull/2312 goes in
- contract PR to go through & contract registered (addresses currently hard-coded)

### CODE REVIEW

File: js/src/dapps/signaturereg/IdentityIcon/identityIcon.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import React, { Component, PropTypes } from 'react';
+
+import { api } from '../parity';
+import styles from './identityIcon.css';
+
+export default class IdentityIcon extends Component {
```
Comment (derhuerst) [inline js/src/dapps/signaturereg/IdentityIcon/identityIcon.js]: Why not use `src/ui/IdentityIcon`?

Comment (jacogr) [inline js/src/dapps/signaturereg/IdentityIcon/identityIcon.js]: Dapps should not rely on the ui/* stuff - we don't build those as a module and would make splitting of the apps from the repo problematic. However, there is probably something to say fro a `dapps-ui` thing that has all the common components. (e.g. for that IdentityIcon the same code is not in Gavcoin, TokenReg & here - that I know of)

So valid concern, just a different issue.

File: js/src/contracts/registry.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import * as abis from '../json';
+
+export default class Registry {
```
Comment (jacogr) [inline js/src/contracts/registry.js]: On purpose, to lessen congnitive overload. Use classes everywhere, so didn't deviate. I actually realised that with the 3rdparty stuff where we don't, it takes a couple of mins to get into it when changes are to be made.

File: js/src/dapps/signaturereg/IdentityIcon/identityIcon.js (hunk 1 above)
Comment (jacogr) [inline js/src/dapps/signaturereg/IdentityIcon/identityIcon.js]: Logged https://github.com/ethcore/parity/issues/2321

File: js/src/ui/MethodDecoding/methodDecoding.css [hunk 3]
```diff
@@ -16,6 +16,7 @@
 */
 
 .method {
+  font-family: 'Roboto Mono' !important;
```
Comment (derhuerst) [inline js/src/ui/MethodDecoding/methodDecoding.css]: Should be `font-family: …, monospace;` to tell the computer that you want monospace event if loading the font fails.

File: js/src/views/Account/Transactions/Transaction/transaction.js [hunk 4]
```diff
@@ -160,7 +161,14 @@ export default class Transaction extends Component {
...
     const ether = api.util.fromWei(value);
 
-    return `${ether.toFormat(5)}`;
+    if (ether.eq(0)) {
+      return { value: null, token: null };
+    }
+
+    return {
+      value: `${ether.toFormat(5)}`,
+      token: <small>ΞTH</small>
```
Comment (derhuerst) [inline js/src/views/Account/Transactions/Transaction/transaction.js]: being really picky – As of HTML5, `<small>` has a semantical meaning and should not be used for stylistic purposes. Use `<span`.

(also mentioned in https://github.com/ethcore/parity/pull/2099#discussion_r79003278)

File: js/src/ui/MethodDecoding/methodDecoding.css [hunk 5]
```diff
@@ -15,11 +15,23 @@
...
+}
+
+.description {
+  line-height: 1.5em;
+  padding-bottom: 0.5em;
+}
+
+.name,
+.inputs {
+  font-family: 'Roboto Mono';
```
Comment (derhuerst) [inline js/src/ui/MethodDecoding/methodDecoding.css]: Should be `font-family: …, monospace;` to tell the computer that you want monospace event if loading the font fails.

File: js/src/ui/MethodDecoding/methodDecoding.css (hunk 3 above)
Comment (jacogr) [inline js/src/ui/MethodDecoding/methodDecoding.css]: Was discussing it earlier today, really missing Sass variables, these things aren't very DRY. Should probably fix it all-over. Which brings me to the next point, fonts should be bundles as part of Webpack. (Tomasz did it somewhere in one of his apps, everything should be served with Parity, fast, no connection needed to play locally)

https://github.com/ethcore/parity/issues/2344

File: js/src/views/Account/Transactions/Transaction/transaction.js (hunk 4 above)
Comment (jacogr) [inline js/src/views/Account/Transactions/Transaction/transaction.js]: Agreed. As commented there, we need a refactor of these completely - I picked up that habit from the Signer code. In addition, we actually are repeating these things everywhere (not just here), really not a good DRY approach.

File: js/src/dapps/signaturereg/Import/import.css [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+.info textarea {
+  background: rgba(80, 80, 80, 0.25);
+  border-radius: 5px;
+  resize: none;
+  padding: 1em;
+  color: #eee;
+  font-size: 0.75em;
+  font-family: 'Roboto Mono';
```
Comment (derhuerst) [inline js/src/dapps/signaturereg/Import/import.css]: My grumble from https://github.com/ethcore/parity/pull/2313#discussion_r80468994 still applies here afaict.

Comment (jacogr) [inline js/src/dapps/signaturereg/Import/import.css]: Until we have the fonts included, then not an issue anymore. Won't fix.

**Pull request merged**